app/controllers/avo/actions_controller.rb

```diff
@@ -104,25 +104,33 @@ def respond
             frame_id = Avo::ACTIONS_TURBO_FRAME_ID
             src, _ = @response[:action].link_arguments(resource: @action.resource, **@response[:navigate_to_action_args])
 
-            render turbo_stream: turbo_stream.turbo_frame_set_src(frame_id, src)
+            turbo_stream.turbo_frame_set_src(frame_id, src)
           when :redirect
-            render turbo_stream: turbo_stream.redirect_to(
+            turbo_stream.redirect_to(
               Avo::ExecutionContext.new(target: @response[:path]).handle,
               turbo_frame: @response[:redirect_args][:turbo_frame],
               **@response[:redirect_args].except(:turbo_frame)
             )
           when :close_modal
             # Close the modal and flash the messages
-            render turbo_stream: [
+            [
               turbo_stream.close_action_modal,
               turbo_stream.flash_alerts
             ]
           else
             # Reload the page
             back_path = request.referer || params[:referrer].presence || resources_path(resource: @resource)
 
-            render turbo_stream: turbo_stream.redirect_to(back_path)
+            turbo_stream.redirect_to(back_path)
           end
+
+          responses = if @action.enhanced_turbo_streams.present?
```

Comment:
can't we include this `if` statement in the `case` statement above?

Reply:
The scope of this feature is to add one or several turbo-stream responses to the pre-existing response

```ruby
def handle(**args)
    succeed "Modal closed!!"
    close_modal
    enhance_response turbo_stream: -> {
      turbo_stream.set_title("Dummy action custom stream ;)")
    }
  end
```

For example on the code above we want the success message to be rendered, the modal to be closed then enhance that response with the `turbo_stream.set_title("Dummy action custom stream ;)")`

If we add the if statement on the case above the action would respond only with `turbo_stream.set_title("Dummy action custom stream ;)")` ignoring `succeed "Modal closed!!"` and `close_modal` responses.